temporalio/exceptions.py

```diff
@@ -71,7 +99,18 @@ def __init__(
 
 
 class ApplicationError(FailureError):
-    """Error raised during workflow/activity execution."""
+    """Error raised during workflow/activity execution to cause a workflow execution failure.
+
+    Workflow Execution Failures put the Workflow Execution into the "Failed" state and no further attempt will
+    be made to progress their execution.
+
+    # Example
+
+    >>> from temporalio.exceptions import ApplicationError
+    ... # ...
+    ... if isDelivery and distance.get_kilometers() > 25:
+    ...     raise ApplicationError("Customer lives outside the service area")
+    """
```

Comment:
Formatted for 80-col width; we don't use the Workflow Execution style capitalization of the Temporal docs in the codebase; snake_case Python variable name.

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block

```suggestion
    """Raised in workflow/activity code to cause a workflow execution failure.

    Workflow execution failure puts the workflow execution into "Failed" state,
    and no further attempts will be made to progress the workflow execution.

    .. code-block:: python

        from temporalio.exceptions import ApplicationError
        # ...
        if is_delivery and distance.get_kilometers() > 25:
            raise ApplicationError("Customer lives outside the service area")
    """
```